lib/FindTests.js

```diff
@@ -0,0 +1,263 @@
+// @flow
+
+const gracefulFs = require('graceful-fs');
+const fs = require('fs');
+const glob = require('glob');
+const path = require('path');
+const Parser = require('./Parser');
+const Project = require('./Project');
+
+void Project;
+
+function flatMap/*:: <T, U> */(
+  array /*: Array<T> */,
+  f /*: (T) => Array<U> */
+) /*: Array<U> */ {
+  return array.reduce((result, item) => result.concat(f(item)), []);
+}
+
+// Double stars at the start and end is the correct way to ignore directories in
+// the `glob` package.
+// https://github.com/isaacs/node-glob/issues/270#issuecomment-273949982
+// https://github.com/isaacs/node-glob/blob/f5a57d3d6e19b324522a3fa5bdd5075fd1aa79d1/common.js#L222-L231
+const ignoredDirsGlobs = ['**/elm-stuff/**', '**/node_modules/**'];
+
+function resolveGlobs(
+  fileGlobs /*: Array<string> */,
+  projectRootDir /*: string */
+) /*: Array<string> */ {
+  return Array.from(
+    new Set(
+      flatMap(fileGlobs, (fileGlob) => {
+        const absolutePath = path.resolve(fileGlob);
+        try {
+          const stat = fs.statSync(absolutePath);
+          // If the CLI arg exists…
+          return stat.isDirectory()
+            ? // …and it’s a directory, find all .elm files in there…
+              findAllElmFilesInDir(absolutePath)
+            : // …otherwise use it as-is.
+              [absolutePath];
+        } catch (error) {
+          // If the CLI arg does not exist…
+          return error.code === 'ENOENT'
+            ? // …resolve it as a glob for shells that don’t support globs.
+              resolveCliArgGlob(absolutePath, projectRootDir)
+            : // The glob package ignores other types of stat errors.
+              [];
+        }
+      })
+    ),
+    // The `glob` package returns absolute paths with slashes always, even on
+    // Windows. All other paths in elm-test use the native directory separator
+    // so normalize here.
+    (filePath) => path.normalize(filePath)
+  );
+}
+
+function resolveCliArgGlob(
+  fileGlob /*: string */,
+  projectRootDir /*: string */
+) /*: Array<string> */ {
+  return flatMap(
+    // Globs passed as CLI arguments are relative to CWD, while elm-test
+    // operates from the project root dir.
+    glob.sync(path.relative(projectRootDir, path.resolve(fileGlob)), {
```

Comment:
```suggestion
    const globRelativeToProjectRoot = path.relative(projectRootDir, path.resolve(fileGlob);
    glob.sync(globRelativeToProjectRoot), {
```

Just makes it clearer that the comment refers the the glob we pass to `glob.sync` rather than the `glob.sync` call in general.